esmvaltool/utils/cmorizers/obs/utilities.py

```diff
@@ -136,15 +140,25 @@ def _roll_cube_data(cube, shift, axis):
     return cube
 
 
-def _save_variable(cube, var, outdir, year, proj):
+def _save_variable(cube, var, outdir, proj, zip='False'):
     """Saver function."""
     # CMOR standard
-    time_suffix = '-'.join([str(year) + '01', str(year) + '12'])
+    yrbeg=cube.coord('time').cell(0).point.strftime('%Y')
+    mobeg=cube.coord('time').cell(0).point.strftime('%m')
+    yrend=cube.coord('time').cell(-1).point.strftime('%Y')
+    moend=cube.coord('time').cell(-1).point.strftime('%m')
+    time_suffix = '-'.join([yrbeg + mobeg, yrend + moend])
+
     cmor_prefix = '_'.join([
         'OBS', proj['dataset'], proj['realm'], proj['version'],
-        proj['frequency'][var], var
+        proj['field'], var, proj['frequency'][var]
     ])
     file_name = cmor_prefix + '_' + time_suffix + '.nc'
     file_path = os.path.join(outdir, file_name)
-    logger.info('Saving: %s', file_path)
-    iris.save(cube, file_path)
+    fillvalue=cube.data.fill_value
+    if ( zip ) :
+        logger.info('Saving as NC4-ZIP : %s ', file_path)
+        iris.save(cube, file_path, fill_value=fillvalue, local_keys=['positive'], zlib=True, complevel=4)
+    else : 
+        logger.info('Saving: %s', file_path)
```

Comment:
we are not supporting NC4 formats so please dont add it here, if you want to save the file in nc4 format do that at the cmorizer script level although I'd encourage it not to

Reply:
My bad I was not aware of that.
I'll revert this change and define a `_save_variable` function within the cmorizer script.